modules/icinga/manifests/host.pp

```diff
@@ -1,10 +1,39 @@
-# FIXME: This class needs better documentation as per https://docs.puppetlabs.com/guides/style_guide.html#puppet-doc
+# == Define: icinga::host
+#
+# Creates a host definition as per:
+# http://docs.icinga.org/latest/en/objectdefinitions.html#host
+#
+# === Parameters:
+#
+#  [*hostalias*]
+#    The title of the host within Icinga, usually `$::fqdn`.
+#
+#
+#  [*address*]
+#    The IP address which is used to report into Icinga, usually
+#    `$::ipaddress`.
+#
+#  [*use*]
+#    FIXME: Cannot find documentation on this parameter for host definitions.
```

Comment:
The `use` parameter in Icinga is to do with [which object it should inherit from](http://docs.icinga.org/latest/en/objectinheritance.html); `generic-host` is a host provided by the Icinga package which lives in `/etc/icinga/conf.d/generic-host_nagios2.cfg` and specifies a bunch of defaults.